src/debugAdapter/goDebug.ts

```diff
@@ -519,6 +517,19 @@ class GoDebugSession extends DebugSession {
 		if (/^(\w:[\\/]|\\\\)/.test(path)) return '\\';
 		return path.includes('/') ? '/' : '\\';
 	}
+	protected convertToHex(v) {
+		let s = v;
+		if (this.numAsHex === 'hex') {
+			s = '0x' + parseInt(v).toString(16);
```

Comment:
This will truncate floating-point numbers.  Perhaps this functionality should be limited to integers?